sdk/rust/src/api.rs

```diff
@@ -11,11 +11,21 @@ pub trait KMS {
         padding_mode: Option<&str>,
         digest_mode: Option<&str>) -> Result<String>;
 
+    async fn get_publickey(&mut self, _keyid: &str) -> Result<String>;
+    async fn sign(&mut self, _keyid: &str, _digest: &str) -> Result<String>;
+    
+    async fn verify(&mut self, _keyid: &str, _digest: &str, _signature: &str) -> Result<bool>;
+
     async fn decrypt(&mut self, _keyid: &str, _data: &str, _aad: Option<&str>) -> Result<String>;
 
     async fn encrypt(&mut self, _keyid: &str, _data: &str, _aad: Option<&str>) -> Result<String>;
+    
+    async fn generate_datakey_without_plaintext(&mut self, _keyid: &str, _len: &i32, _aad: Option<&str>)-> Result<String>;
+    async fn generate_datakey(&mut self, _keyid: &str, _len: &i32, _aad: Option<&str>)-> Result<String>;
+    async fn asymmetric_encrypt(&mut self, _keyid: &str, _data: &str)-> Result<String>;
+    async fn asymmetric_decrypt(&mut self, _keyid: &str, _data: &str)-> Result<String>;
+    async fn export_datakey(&mut self, _keyid: &str, _ukeyid: &str, _datakey: &str, _aad: Option<&str>)-> Result<String>;
```

Comment:
Is the returned value the base64-encoded datakey?

Reply:
yes, it should be